### PR TITLE
linux

### PR DESCRIPTION
**Pull request by ocefpaf**
@Korijn this is currently failing to find `libwx_gtk2u_core-3.0.so.0` on linux at the import test. But the file does exist :unamused: 

```shell
+ source /opt/conda/bin/activate /feedstock_root/build_artefacts/wxpython_1498486398574/_t_env
+ /feedstock_root/build_artefacts/wxpython_1498486398574/_t_env/bin/python -s /feedstock_root/build_artefacts/wxpython_1498486398574/test_tmp/run_test.py
import: 'wx'
Traceback (most recent call last):
  File "/feedstock_root/build_artefacts/wxpython_1498486398574/test_tmp/run_test.py", line 2, in <module>
    import wx
  File "/feedstock_root/build_artefacts/wxpython_1498486398574/_t_env/lib/python2.7/site-packages/wx/__init__.py", line 17, in <module>
    from wx.core import *
  File "/feedstock_root/build_artefacts/wxpython_1498486398574/_t_env/lib/python2.7/site-packages/wx/core.py", line 12, in <module>
    from ._core import *
ImportError: libwx_gtk2u_core-3.0.so.0: cannot open shared object file: No such file or directory
TESTS FAILED: wxpython-4.0.0a2-py27_0.tar.bz2
```

Not sure what happens on `OS X`, so I activate it here too just to test.

### CODE REVIEW

**Comment by Korijn** [comment]
I hope it's not an upstream issue.

**Comment by Korijn** [comment]
Seems related, with an answer from Robin (wx maintainer):

https://stackoverflow.com/a/31928092/552379

**Comment by Korijn** [comment]
Aw! Timeouts!

```
[744/858] cxx: sip/cpp/sip_propgridwxPGArrayStringEditorDialog.cpp -> build/waf/3.6/sip/cpp/sip_propgridwxPGArrayStringEditorDialog.cpp.29.o



The job exceeded the maximum time limit for jobs, and has been terminated.
```

I think it means you did also figure out the Mac OS build though. :) Awesome!

**Comment by Korijn** [comment]
Hope this helps: https://github.com/conda-forge/conda-smithy/pull/539/files

**Comment by ocefpaf** [comment]
> I think it means you did also figure out the Mac OS build though. :) Awesome!

Not sure b/c I did not build this locally. I don't own a Mac. And Travis-CI never got to the end here :smile: 

> Hope this helps: https://github.com/conda-forge/conda-smithy/pull/539/files

Looks interesting and definitely useful here and for `ncl-feedstock`.

Note that, IMO, I would like to see things like `yum_requirements.txt` go away instead of being widely adopt. These are workarounds that can make our life complicated in the long run. This is not a vote against the conda-smithy PR! We do need that for special cases like this one. It is more a concern regarding its use.

**Comment by Korijn** [comment]
@ocefpaf @jakirkham how do we move forward?

**Comment by ocefpaf** [comment]
@Korijn I am swamped with work at the moment and pretty soon I will be out of office for a few weeks. I'll be able to get back to this only in August.

**Comment by Korijn** [comment]
OK. Any tips for others who might want to pursue this avenue?

**Comment by ocefpaf** [comment]
I guess that not much besides that SO link you found :-(

**********************************************************
Filipe P. A. Fernandes
Physical Oceanographer

Email: ocefpaf@gmail.com

http://ocefpaf.github.io/
**********************************************************

**Comment by Korijn** [comment]
Locally, `ldd` gives me the following output:

```
(/media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env) korijn@swatcat:/media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/work$ ldd /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/libwx_gtk2u_core-3.0.so.0
	linux-vdso.so.1 =>  (0x00007ffe81947000)
	libgtk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0 (0x00007efd5d3f2000)
	libgdk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0 (0x00007efd5d13d000)
	libpangocairo-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0 (0x00007efd5cf30000)
	libcairo.so.2 => /usr/lib/x86_64-linux-gnu/libcairo.so.2 (0x00007efd5cc14000)
	libgdk_pixbuf-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0 (0x00007efd5c9f1000)
	libpango-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0 (0x00007efd5c7a3000)
	libgobject-2.0.so.0 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../../libgobject-2.0.so.0 (0x00007efd5c54e000)
	libglib-2.0.so.0 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../../libglib-2.0.so.0 (0x00007efd5c223000)
	libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007efd5beea000)
	libXxf86vm.so.1 => /usr/lib/x86_64-linux-gnu/libXxf86vm.so.1 (0x00007efd5bce4000)
	libSM.so.6 => /usr/lib/x86_64-linux-gnu/libSM.so.6 (0x00007efd5badc000)
	libpng16.so.16 => /usr/lib/x86_64-linux-gnu/libpng16.so.16 (0x00007efd5b8a8000)
	libz.so.1 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../../libz.so.1 (0x00007efd5b68b000)
	libwx_baseu-3.0.so.0 => not found
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007efd5b382000)
	libstdc++.so.6 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../../libstdc++.so.6 (0x00007efd5b002000)
	libgcc_s.so.1 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../../libgcc_s.so.1 (0x00007efd5adec000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007efd5abcc000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007efd5a805000)
	libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007efd5a601000)
	libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007efd5a3fb000)
	libatk-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0 (0x00007efd5a1d6000)
	libgio-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007efd59e41000)
	libpangoft2-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0 (0x00007efd59c2a000)
	libfontconfig.so.1 => /usr/lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007efd599e7000)
	libXrender.so.1 => /usr/lib/x86_64-linux-gnu/libXrender.so.1 (0x00007efd597dd000)
	libXinerama.so.1 => /usr/lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007efd595da000)
	libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007efd593ca000)
	libXrandr.so.2 => /usr/lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007efd591bf000)
	libXcursor.so.1 => /usr/lib/x86_64-linux-gnu/libXcursor.so.1 (0x00007efd58fb3000)
	libXcomposite.so.1 => /usr/lib/x86_64-linux-gnu/libXcomposite.so.1 (0x00007efd58db0000)
	libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007efd58bad000)
	libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007efd5899b000)
	libfreetype.so.6 => /usr/lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007efd586ee000)
	libpixman-1.so.0 => /usr/lib/x86_64-linux-gnu/libpixman-1.so.0 (0x00007efd58448000)
	libxcb-shm.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0 (0x00007efd58242000)
	libxcb-render.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-render.so.0 (0x00007efd58038000)
	libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007efd57e16000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007efd57c0e000)
	libthai.so.0 => /usr/lib/x86_64-linux-gnu/libthai.so.0 (0x00007efd57a05000)
	libiconv.so.2 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../.././libiconv.so.2 (0x00007efd57715000)
	libpcre.so.1 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../.././libpcre.so.1 (0x00007efd574d3000)
	libffi.so.6 => /media/ssd2/miniconda3/conda-bld/wxpython_1499089280147/_t_env/lib/python3.6/site-packages/wx/../../.././libffi.so.6 (0x00007efd572ca000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007efd570c6000)
	libICE.so.6 => /usr/lib/x86_64-linux-gnu/libICE.so.6 (0x00007efd56eac000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007efd56ca5000)
	/lib64/ld-linux-x86-64.so.2 (0x000055e2373ac000)
	libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007efd56a7d000)
	libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007efd56862000)
	libmount.so.1 => /lib/x86_64-linux-gnu/libmount.so.1 (0x00007efd56617000)
	libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007efd56383000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007efd56157000)
	libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007efd55f53000)
	libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007efd55d4d000)
	libdatrie.so.1 => /usr/lib/x86_64-linux-gnu/libdatrie.so.1 (0x00007efd55b46000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007efd558d3000)
	libblkid.so.1 => /lib/x86_64-linux-gnu/libblkid.so.1 (0x00007efd5568c000)
	libgraphite2.so.3 => /usr/lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007efd55466000)
```

**Comment by Korijn** [comment]
The OS X build actually produces a package that works, although these two test commands fail:

```
     - conda inspect linkages -p $PREFIX wxPython  # [not win]
     - conda inspect objects -p $PREFIX wxPython  # [osx]
```

It is missing a runtime dependency `python.app` though, to get `pythonw` on the command line.

If we can address that, we have a working build and a working package for OS X. Just a timeout issue that remains.

**Comment by ocefpaf** [comment]
> The OS X build actually produces a package that works, although these two test commands fail:

So the conda-inspect is causing the time-out? We can change that to:

```
     - conda inspect linkages -p $PREFIX wxPython  # [linux]
```

Let me push that here...

**Comment by Korijn** [comment]
No, it's not. It's timing out before the build passes.

But the build passes on a local OS X machine we have in the office here. :)

**Comment by ocefpaf** [comment]
> No, it's not. It's timing out before the build passes.

Ah... So unfortunately there is not much we can do here :unamused: 

> But the build passes on a local OS X machine we have in the office here. :)

I wonder why the conda-inspect failed for you there.. It should pass.

**Comment by Korijn** [comment]
So do I! The code does actually work just fine... I can run wx code with the package.

**Comment by ocefpaf** [comment]
> So do I! The code does actually work just fine... I can run wx code with the package.

What is the error? If we investigate this we and figure out what is getting linked we can use try to upload your package to the channel.

**Comment by Korijn** [comment]
I don't have the exact traceback here, but it read something like "Package wxPython is not installed in {prefix here}"

**Comment by ocefpaf** [comment]
> but it read something like "Package wxPython is not installed in {prefix here}"

This may be a conda issue. What are the conda and conda-build version used?

**Comment by berendkleinhaneveld** [comment]
```
Current conda install:

               platform : osx-64
          conda version : 4.3.22
       conda is private : False
      conda-env version : 4.3.22
    conda-build version : 2.1.15
         python version : 3.6.1.final.0
       requests version : 2.14.2
       root environment : /Users/beer/miniconda3  (writable)
    default environment : /Users/beer/miniconda3
       envs directories : /Users/beer/miniconda3/envs
                          /Users/beer/.conda/envs
          package cache : /Users/beer/miniconda3/pkgs
                          /Users/beer/.conda/pkgs
           channel URLs : https://conda.anaconda.org/clinicalgraphics/osx-64
                          https://conda.anaconda.org/clinicalgraphics/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /Users/beer/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.22 requests/2.14.2 CPython/3.6.1 Darwin/16.6.0 OSX/10.12.5    
                UID:GID : 501:20```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,20 +13,32 @@ source:
...
+    - gst-plugins-base  # [linux]
+    - mesa  # [linux]
+    - libglu  # [linux]
   run:
     - python
     - six
+    - gstreamer  # [linux]
+    - gst-plugins-base  # [linux]
+    - mesa  # [linux]
+    - libglu  # [linux]
```
Comment (Korijn) [inline recipe/meta.yaml]: Can you add `python.app  # [osx]`? You need `pythonw` to run wx apps on OS X.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -13,20 +13,32 @@ source:
...
 
 test:
   imports:
     - wx
+  commands:
+    - conda inspect linkages -p $PREFIX wxPython  # [not win]
+    - conda inspect objects -p $PREFIX wxPython  # [osx]
```
Comment (Korijn) [inline recipe/meta.yaml]: This currently fails on OSX. Not sure why.

**Comment by Korijn** [comment]
@ocefpaf see Berend's post for the conda version used. Can you work with that? Do you need something from me?

**Comment by ocefpaf** [comment]
> @ocefpaf see Berend's post for the conda version used

Yep. I don't see anything wrong with that... Not sure what is going on.

**Comment by Korijn** [comment]
As it turns out, wxPython does its own magic altering paths in binaries to make the installation portable. The author suggests we disable that behaviour for this build, so that conda can do its own thing.

>>> I would try it first as 2 separate commands, `python build.py build_wx install_wx --no_magic` and `python build.py build_py install_py --no_magic --use_syswx`

Source: https://github.com/wxWidgets/Phoenix/issues/423

**Comment by ocefpaf** [comment]
> As it turns out, wxPython does its own magic altering paths

Aha! I will try that now... Hold on...

**Comment by Korijn** [comment]
The CircleCI build log won't load for me... annoying much...

**Comment by ocefpaf** [comment]
> The CircleCI build log won't load for me... annoying much...

I had a similar issue in the past with chromium. Did you try another browser?

Anyways, the log is truncated because we hit: `This output was truncated to stay under the 4MB limit.`

I am adding a workaround for that, but locally I still have the same issue at the import test.

**Comment by Korijn** [comment]
There apparently also is a `--prefix` switch we can try.

File: recipe/build.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+# If an error occurs, run our error handler to output a tail of the build.
+trap 'error_handler' ERR
+
+# Set up a repeating loop to send some output to Travis.
+bash -c "while true; do echo \$(date) - building ...; sleep $PING_SLEEP; done" &
+PING_LOOP_PID=$!
+
+## START BUILD
```
Comment (ocefpaf) [inline recipe/build.sh]: @Korijn all this boilerplate is just to keep pinging, so Travis-CI does not quit due to lack of response or the 4 MB log limit while also saving the log in the `$BUILD_OUTPUT`.

We are not compiling on OS X for now but I wanted to put this here anyways.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,32 +1,42 @@
-{% set name = "wxPython" %}
-{% set version = "4.0.0a2" %}
-{% set sha256 = "3acd5160d69f56b588476d1907eff67b53ad58fb2eba4dedabcdc3490d170338" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml]: This is more readable IMO and I have a script that updates the recipe automatically when the only jinja var is the version number.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,32 +1,42 @@
...
-  sha256: {{ sha256 }}
+  fn: wxpython-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/w/wxpython/wxPython-{{ version }}.tar.gz
+  sha256: b147b1a3b4d10da88100fc89a3c4225ba6ff1f89e5da77218989b198ab92b2e6
 
 build:
   number: 0
-  script: python setup.py install --single-version-externally-managed --record record.txt
-  skip: True  # [unix]
+  skip: True  # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: This recipe should work on `OS X` now but `Travis-CI` will time-out.
We need volunteers to build it.

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [inline recipe/meta.yaml]: Done!

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,32 +1,42 @@
...
+    - gst-plugins-base  # [linux]
+    - mesa  # [linux]
+    - libglu  # [linux]
 
 test:
   imports:
     - wx
+  commands:
+    - conda inspect linkages -p $PREFIX wxpython  # [not win]
+    - conda inspect objects -p $PREFIX wxpython  # [osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: The issue turned out to be the package name. That is why I don't like to jinjify everything, it makes it harder to debug things.

File: recipe/yum_requirements.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+gtk2-devel
```
Comment (ocefpaf) [inline recipe/yum_requirements.txt]: I don't really like this b/c we end up with a package that links to a lot of system stuff. We need a `gtk` package on `conda-forge`. I am looking into it...

File: recipe/meta.yaml (hunk 6 above)
Comment (Korijn) [inline recipe/meta.yaml]: shame about the timeouts on travis. we actually have a working OSX build now too, then!

File: recipe/yum_requirements.txt (hunk 7 above)
Comment (Korijn) [inline recipe/yum_requirements.txt]: Awesome.

File: recipe/meta.yaml (hunk 6 above)
Comment (ocefpaf) [inline recipe/meta.yaml]: Yep. If you can find a OS X machine and build it we can try a manual upload (we do that already for Qt).

Comment (Korijn) [inline recipe/meta.yaml]: @berendkleinhaneveld would you like to volunteer? :)

**Comment by ocefpaf** [comment]
`conda-inspect` results:

```
system:
    libGL.so.1 (/usr/lib64/libGL.so.1)
    libICE.so.6 (/usr/lib64/libICE.so.6)
    libSM.so.6 (/usr/lib64/libSM.so.6)
    libX11-xcb.so.1 (/usr/lib64/libX11-xcb.so.1)
    libX11.so.6 (/usr/lib64/libX11.so.6)
    libXau.so.6 (/usr/lib64/libXau.so.6)
    libXcomposite.so.1 (/usr/lib64/libXcomposite.so.1)
    libXcursor.so.1 (/usr/lib64/libXcursor.so.1)
    libXdamage.so.1 (/usr/lib64/libXdamage.so.1)
    libXext.so.6 (/usr/lib64/libXext.so.6)
    libXfixes.so.3 (/usr/lib64/libXfixes.so.3)
    libXi.so.6 (/usr/lib64/libXi.so.6)
    libXinerama.so.1 (/usr/lib64/libXinerama.so.1)
    libXrandr.so.2 (/usr/lib64/libXrandr.so.2)
    libXrender.so.1 (/usr/lib64/libXrender.so.1)
    libXt.so.6 (/usr/lib64/libXt.so.6)
    libXxf86vm.so.1 (/usr/lib64/libXxf86vm.so.1)
    libatk-1.0.so.0 (/usr/lib64/libatk-1.0.so.0)
    libc.so.6 (/lib64/libc.so.6)
    libcairo.so.2 (/usr/lib64/libcairo.so.2)
    libdl.so.2 (/lib64/libdl.so.2)
    libdrm.so.2 (/usr/lib64/libdrm.so.2)
    libenchant.so.1 (/usr/lib64/libenchant.so.1)
    libexpat.so.1 (/lib64/libexpat.so.1)
    libfontconfig.so.1 (/usr/lib64/libfontconfig.so.1)
    libfreetype.so.6 (/usr/lib64/libfreetype.so.6)
    libgailutil.so.18 (/usr/lib64/libgailutil.so.18)
    libgcc_s.so.1 (/lib64/libgcc_s.so.1)
    libgdk-x11-2.0.so.0 (/usr/lib64/libgdk-x11-2.0.so.0)
    libgdk_pixbuf-2.0.so.0 (/usr/lib64/libgdk_pixbuf-2.0.so.0)
    libglapi.so.0 (/usr/lib64/libglapi.so.0)
    libgstapp-0.10.so.0 (/usr/lib64/libgstapp-0.10.so.0)
    libgstbase-0.10.so.0 (/usr/lib64/libgstbase-0.10.so.0)
    libgstinterfaces-0.10.so.0 (/usr/lib64/libgstinterfaces-0.10.so.0)
    libgstpbutils-0.10.so.0 (/usr/lib64/libgstpbutils-0.10.so.0)
    libgstreamer-0.10.so.0 (/usr/lib64/libgstreamer-0.10.so.0)
    libgstvideo-0.10.so.0 (/usr/lib64/libgstvideo-0.10.so.0)
    libgtk-x11-2.0.so.0 (/usr/lib64/libgtk-x11-2.0.so.0)
    libicudata.so.42 (/usr/lib64/libicudata.so.42)
    libicui18n.so.42 (/usr/lib64/libicui18n.so.42)
    libicuuc.so.42 (/usr/lib64/libicuuc.so.42)
    libjpeg.so.62 (/usr/lib64/libjpeg.so.62)
    libm.so.6 (/lib64/libm.so.6)
    libpango-1.0.so.0 (/usr/lib64/libpango-1.0.so.0)
    libpangocairo-1.0.so.0 (/usr/lib64/libpangocairo-1.0.so.0)
    libpangoft2-1.0.so.0 (/usr/lib64/libpangoft2-1.0.so.0)
    libpixman-1.so.0 (/usr/lib64/libpixman-1.so.0)
    libpng12.so.0 (/usr/lib64/libpng12.so.0)
    libpthread.so.0 (/lib64/libpthread.so.0)
    libresolv.so.2 (/lib64/libresolv.so.2)
    librt.so.1 (/lib64/librt.so.1)
    libselinux.so.1 (/lib64/libselinux.so.1)
    libsoup-2.4.so.1 (/usr/lib64/libsoup-2.4.so.1)
    libstdc++.so.6 (/usr/lib64/libstdc++.so.6)
    libutil.so.1 (/lib64/libutil.so.1)
    libuuid.so.1 (/lib64/libuuid.so.1)
    libwebkitgtk-1.0.so.0 (/usr/lib64/libwebkitgtk-1.0.so.0)
    libxcb-dri2.so.0 (/usr/lib64/libxcb-dri2.so.0)
    libxcb-glx.so.0 (/usr/lib64/libxcb-glx.so.0)
    libxcb.so.1 (/usr/lib64/libxcb.so.1)
    libxml2.so.2 (/usr/lib64/libxml2.so.2)
    libxslt.so.1 (/usr/lib64/libxslt.so.1)
    linux-vdso.so.1 ()
```

a package that depends on so many system libs is very brittle. We need to fix that...

File: recipe/meta.yaml (hunk 6 above)
Comment (berendkleinhaneveld) [inline recipe/meta.yaml]: On it!

**Comment by Korijn** [comment]
Why are there system deps like gstreamer though? Didnt you add a conda dep for that?

**Comment by ocefpaf** [comment]
> Why are there system deps like gstreamer though? Didnt you add a conda dep for that?

When we install `gtk2-devel` and `webkitgtk-devel` with `yum` we make them available and `wx` ends up linking to them. The best solution would be to package those two.

File: recipe/meta.yaml (hunk 6 above)
Comment (berendkleinhaneveld) [inline recipe/meta.yaml]: Too bad, the test failed on `import wx`. I've created a gist with the complete [build output](https://gist.github.com/berendkleinhaneveld/f5f1b68f89f086cb459f392d3b3b74f9).

Comment (Korijn) [inline recipe/meta.yaml]: Weird... why would it try to load a file from the work dir?

File: recipe/meta.yaml [hunk 8]
```diff
@@ -1,32 +1,42 @@
...
+  fn: wxpython-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/w/wxpython/wxPython-{{ version }}.tar.gz
+  sha256: b147b1a3b4d10da88100fc89a3c4225ba6ff1f89e5da77218989b198ab92b2e6
 
 build:
   number: 0
-  script: python setup.py install --single-version-externally-managed --record record.txt
-  skip: True  # [unix]
+  skip: True  # [osx]
+  script: python setup.py install --single-version-externally-managed --record record.txt  # [win]
```
Comment (Korijn) [inline recipe/meta.yaml]: So actually, this build command produced a working package on OS X. Can we make it `win or osx`?

Comment (Korijn) [inline recipe/meta.yaml]: @ocefpaf ?

Comment (ocefpaf) [inline recipe/meta.yaml]: Makes sense. Let me try that. (Sorry for the delay, got lost clearing the e-mail backlog from SciPy.)

Comment (Korijn) [inline recipe/meta.yaml]: no worries

**Comment by Korijn** [comment]
* I think you're missing the `brew install xquartz` line
* You're not using `python setup.py install --single-version-externally-managed --record record.txt` which actually works on OS X in @berendkleinhaneveld's manual builds

**Comment by ocefpaf** [comment]
> * I think you're missing the `brew install xquartz` line
> * You're not using `python setup.py install --single-version-externally-managed --record record.txt` which actually works on OS X in @berendkleinhaneveld's manual builds

Thanks @Korijn! I'll get to it soon. (Had to leave my laptop in the middle of that push to solve smt. Be right back though.)

**Comment by ocefpaf** [comment]
@Korijn as expected `OS X` is timing out and Linux has way too many system libs dependencies. I will try to add `gtk` next week, but if you want to risk this as-is I can add the `OS X` skip and we can merge it.

**Comment by Korijn** [comment]
Let's do that and open a new PR for that step then. Thanks so far, this is great progress!

**Comment by ocefpaf** [comment]
@Korijn I added the OS X skip, feel free to merge whenever you think this ready. I'll work on a `gtk` recipe over the weekend. Thanks for all the help!

**Comment by Korijn** [comment]
Should we do a rerender before mergin then?

**Comment by ocefpaf** [comment]
Yep. Forgot to push. Thanks!

**Comment by ocefpaf** [comment]
@Korijn as soon as this is merged I'll create a branch for the py36+linux build.

**Comment by Korijn** [comment]
Done!

**Pull request merged**